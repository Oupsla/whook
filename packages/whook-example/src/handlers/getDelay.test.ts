import initGetDelay from './getDelay';

describe('getDelay', () => {
  const delay = {
    create: jest.fn(),
    clear: jest.fn(),
  };

  beforeEach(() => {
    delay.create.mockReset();
  });

  it('should work', async () => {
    delay.create.mockResolvedValueOnce(undefined);
    const getDelay = await initGetDelay({
      delay,
    });
    const response = await getDelay({
      duration: 1000,
    });

    expect({
      response,
      delayCreateCalls: delay.create.mock.calls,
      delayClearCalls: delay.clear.mock.calls,
    }).toMatchSnapshot();
  });
});
