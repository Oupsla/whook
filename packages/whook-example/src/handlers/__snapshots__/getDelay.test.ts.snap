// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getDelay should work 1`] = `
Object {
  "delayClearCalls": Array [],
  "delayCreateCalls": Array [
    Array [
      1000,
    ],
  ],
  "response": Object {
    "status": 200,
  },
}
`;
